libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplatesParser.cs

```diff
@@ -186,16 +193,16 @@ private static IParseTree AntlrParseTemplates(string text, string id)
             return parser.file();
         }
 
-        private static IList<Templates> GetReferences(Templates file, Dictionary<string, Templates> cachedTemplates = null)
+        private static IList<Templates> GetReferences(Templates file, Dictionary<string, Templates> cachedTemplates = null, EventHandler registeSourceMap = null)
```

Comment:
I'm not sure why we need eventCallback in parsing process? 

I assume people can take the result of parseTree, and do what ever they want, so event callback is more useful in evaluation pharse than parse phase, isn't it?

Reply:
I know, you mean we provide a new method `registeTemplateAndExpression(EventHandler onEvent)`, in this method, we would go through the parsetree and register all templates and expressions.